includes/Context.php

```diff
@@ -317,4 +349,52 @@ public function is_network_active() {
 
 		return $this->network_active;
 	}
+
+	/**
+	 * Creates the entity for a given post object.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @param \WP_Post $post A WordPress post object.
+	 * @return Entity The entity for the post.
+	 */
+	private function create_entity_for_post( \WP_Post $post ) {
+		$type = 'post';
+
+		// If this post is assigned as the home page, it is actually the blog archive.
+		if ( (int) get_option( 'page_for_posts' ) === (int) $post->ID ) {
+			$type = 'home';
+		}
+
+		return new Entity(
+			$this->filter_reference_url( get_permalink( $post ) ),
+			array(
+				'type'  => $type,
+				'title' => $post->post_title,
+				'id'    => $post->ID,
+			)
+		);
+	}
+
+	/**
+	 * Filters the given URL to ensure the reference URL is used as part of it.
+	 *
+	 * If the site reference URL differs from the home URL (e.g. via filters),
+	 * this method performs the necessary replacement.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @param string $url Input URL.
+	 * @return string URL that starts with the site reference URL.
+	 */
+	private function filter_reference_url( $url ) {
+		$reference_site_url = $this->get_reference_site_url();
+		$orig_site_url      = home_url();
+
+		if ( $orig_site_url !== $reference_site_url ) {
+			$url = str_replace( $orig_site_url, $reference_site_url, $url );
+		}
```

Comment:
I think this method might make more sense if it was responsible for invoking the `googlesitekit_site_url` filter which is currently in `get_reference_site_url()`.

Then `get_reference_site_url()` would simply call `filter_reference_url( home_url() )` and apply the slash. 

`filter_reference_url()` would apply that filter to get the effective URL, and then do the replacement on the given URL, without needing to call `get_reference_site_url()`. Make sense?